Common/Product/SharedProject/HierarchyNode.cs

```diff
@@ -590,7 +590,7 @@ internal abstract class HierarchyNode :
                     break;
 
                 case __VSHPROPID.VSHPROPID_Name:
-                    result = this.Caption;
+                    result = CommonUtils.GetFileOrDirectoryName(this.Url);

```

Comment:
Yeah, I can change those too - most of those (e.g. GetEditLabel) are overridden in NodejsFolderNode, but may as well fix it in HierarcyNode.